server/aesdsocket.c

```diff
@@ -0,0 +1,434 @@
+            perror("server: accept");
+            //closeAll();
+            //exit(EXIT_FAILURE);
+            continue;
+        }
+        else 
+        {
+            syslog(LOG_DEBUG, "CONNECTED");
+        }
+        //syslog(LOG_DEBUG, "new_fd is %d", new_fd);
+        inet_ntop(AF_INET, &(peer_addr.sin_addr), ip4, sizeof(ip4));
+        syslog(LOG_DEBUG, "Accepted connection from %s", ip4);
+
+        addibuffer = 0;
+        //total_size = BUF_SIZE;
+        char *my_buffer = (char *) malloc(BUF_SIZE);
+        if (my_buffer == NULL)
+        {
+            syslog(LOG_ERR, "Failed to malloc.");
+            close(new_fd);
+            exit(EXIT_FAILURE);
+        }
+        // set buffer values all to zero
+        memset(my_buffer, 0, BUF_SIZE);
+
+        do{
+            
+            bytes_rx = recv(new_fd, my_buffer + addibuffer, BUF_SIZE-1, 0); 
+            syslog(LOG_DEBUG, "i have received %ld bytes", bytes_rx);
+            if (bytes_rx < 0)
+            {
+                free(my_buffer);
+                perror("server: recv");
+                syslog(LOG_ERR, "recv: error??");
+                exit(EXIT_FAILURE);
+            }
+            addibuffer += bytes_rx; 
+            new_line = strchr(my_buffer, '\n'); //if NULL, keep getting packets
+            if (new_line != NULL)
+            {
+                syslog(LOG_DEBUG, "YAY, slash n found");
+                isPacketValid = true;
+                addibuffer = new_line - my_buffer + 1; 
+                syslog(LOG_DEBUG, "addibuffer size within slash found %ld", addibuffer);
+            }
+
+            else
+            {
+                my_buffer = realloc(my_buffer, incrementBy+BUF_SIZE);
+                if(my_buffer == NULL)
+                {
+                    syslog(LOG_ERR, "Failed to realloc.");
+                    close(new_fd);
+                    exit(EXIT_FAILURE);
+                }
+                incrementBy += BUF_SIZE;
+                memset(my_buffer+addibuffer, 0 , incrementBy-addibuffer);
+                //addibuffer = 0;
+            }
+        //syslog(LOG_DEBUG, "I am here once only. %d", isPacketValid);
+
+        }while (isPacketValid == false);
+
+        //new line character has been found, now i want to send?
+        //append happens here, writing
+        if(isPacketValid == true)
+        {   
+            //my_buffer[addibuffer]= '\0';
+            //ssize_t write_size = new_line - my_buffer + 1;
+            syslog(LOG_DEBUG, "PACKET SUCCESSFULLY VALIDATED");
+            nr = write(recvfile_fd, my_buffer, addibuffer);
+            syslog(LOG_DEBUG, "nr is %ld",nr);
+            if (nr != addibuffer)
+            {
+                /*error*/
+                int err1 = errno;
+                syslog(LOG_ERR, "Failed to write bytes: errno -> %d", err1);
+                syslog(LOG_ERR, "error string is %s", strerror(errno));
+                exit(EXIT_FAILURE);
```

Comment:
What happens to the allocated memory in this case?
There are multiple places where you forgot to free the memory before you exit.